### DISCUSSION TITLE
Modern Search v4 - June 2022

### DISCUSSION THREAD

**Discussion by FranckyC** [microsoft-search/pnp-modern-search#2193] [Announcements]
# PnP Modern Search v4

**This solution includes now only one `.sppkg` file:**
- `pnp-modern-search-parts-v4.sppkg`. Mandatory.

**Getting started documentation here:** [https://microsoft-search.github.io/pnp-modern-search/](https://microsoft-search.github.io/pnp-modern-search/)

[Feedback and comments](https://github.com/microsoft-search/pnp-modern-search/issues) are welcomed!

> **_NOTE:_ We've updated the SPFx version to `1.14.0`. If you have custom extensions, you must rebuild them using SPFx v1.14.0. If not they will not load.**

### New features

- PR #2116 - Added support of adaptive cards layout rendering and Microsoft Search result types.
- PR #2125 - Allow tokens in "Send the query to a new page/ Property Page URL".
- PR #2135
	* Added a trim duplicates options for SharePoint search and Microsoft Search (beta only).
	* Added support of query template in Microsoft Search (beta endpoint only).

### Fixes

- PR #2049 - Fixed #1803 Security fixes.
- PR #2065 - Fixed #2059 Support SharePoint soft navigation.
- PR #2068, #2043, #2108 - Fixed typos.
- PR #2087 - Fixed #2080 Udpdating the result types Handlebars expression to use the slot function and handle complex property paths (like resource."@odata.type").
- PR #2157 -  Fixed #2151 - Fixed preview pane for 'List' and 'Cards' layouts when item selection is enabled.
- PR #2148 - Sanitize input query in getCountMessage
- PR #2181 - Fix/set graph version header
- PR #2171 - Fixed #2170 rowheader default changed to false
- PR #2167 - Support current URL for urlParse

### Fixes

- PR #2104 - Fixed #2031 Now the filter combo box values list is not re rendered when a filtered value is selected allowing to select multiple values consecutively.
- PR #2119 - Fixed #2072 Issue with localization enabled.

### Fixes 

- PR #2133 - Fixed #2001 Check box filter text color for dark background.
- PR #2132 - Fixed #2130 Added a check on the vertical tab hyperlink if the search box query is empty and pass an empty string instead of undefined.
- PR #2134 - Fixed #2027 Fixed date intervals wording to match actual filter conditions #2027. 

### Fixes 

- PR #2090 - Checking nulls when clearing search.

## Key contributors

- @wobba 
- @FranckyC 
- @d-martins
- @bruzie
- @rick-brown-slalom
- @mrjarbenne
- @davemehr
- @dkardokas

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft-search/pnp-modern-search/releases/tag/4.7.0'>Modern Search v4 - June 2022</a>.</em>

**Comment by davemehr** [comment]
Hello @FranckyC , i have created a new scenario [Create-a-useful-People-Search.md] but i don't see it public. Does i need to do anything to publish the new scenario?

**Reply by FranckyC** [reply]
My mistake, I forgot to publish the latest. Here you go https://microsoft-search.github.io/pnp-modern-search/scenarios/Create-a-useful-People-Search/ :D
